### ISSUE TITLE
assignment 6

### ISSUE THREAD

**Issue by bhagwatsolunke** [bhagwatsolunke/dl#7]
Assignment_6
November 7, 2023
[1]: from tensorflow.keras.utils import load_img
from tensorflow.keras.utils import img_to_array
from keras.applications.vgg16 import preprocess_input
from keras.applications.vgg16 import decode_predictions
from keras.applications.vgg16 import VGG16
[2]: image = load_img('/home/rmdstic/Desktop/Avadhut Tehare/tajmahal.jpg',␣
↪target_size=(224, 224))
[3]: image = img_to_array(image)
[4]: print(image)
[[[ 0. 5. 11.]
[ 1. 6. 12.]
[ 1. 8. 18.]
…
[ 6. 28. 75.]
[ 6. 28. 75.]
[ 6. 28. 75.]]
[[ 0. 5. 11.]
[ 1. 6. 12.]
[ 1. 8. 18.]
…
[ 6. 28. 75.]
[ 6. 28. 75.]
[ 6. 28. 75.]]
[[ 0. 5. 11.]
[ 1. 6. 12.]
[ 1. 8. 18.]
…
[ 6. 28. 75.]
[ 6. 28. 75.]
[ 6. 28. 75.]]
…
1
[[ 0. 5. 9.]
[ 1. 5. 14.]
[ 0. 7. 17.]
…
[ 3. 21. 67.]
[ 5. 22. 68.]
[ 5. 22. 68.]]
[[ 1. 6. 12.]
[ 1. 6. 12.]
[ 0. 7. 17.]
…
[ 3. 20. 64.]
[ 0. 18. 64.]
[ 3. 21. 67.]]
[[ 1. 6. 12.]
[ 1. 6. 12.]
[ 0. 7. 17.]
…
[ 1. 18. 62.]
[ 3. 21. 67.]
[ 5. 23. 69.]]]
[5]: image = image.reshape((1, image.shape[0], image.shape[1], image.shape[2]))
[6]: print(image)
[[[[ 0. 5. 11.]
[ 1. 6. 12.]
[ 1. 8. 18.]
…
[ 6. 28. 75.]
[ 6. 28. 75.]
[ 6. 28. 75.]]
[[ 0. 5. 11.]
[ 1. 6. 12.]
[ 1. 8. 18.]
…
[ 6. 28. 75.]
[ 6. 28. 75.]
[ 6. 28. 75.]]
[[ 0. 5. 11.]
[ 1. 6. 12.]
[ 1. 8. 18.]
2
…
[ 6. 28. 75.]
[ 6. 28. 75.]
[ 6. 28. 75.]]
…
[[ 0. 5. 9.]
[ 1. 5. 14.]
[ 0. 7. 17.]
…
[ 3. 21. 67.]
[ 5. 22. 68.]
[ 5. 22. 68.]]
[[ 1. 6. 12.]
[ 1. 6. 12.]
[ 0. 7. 17.]
…
[ 3. 20. 64.]
[ 0. 18. 64.]
[ 3. 21. 67.]]
[[ 1. 6. 12.]
[ 1. 6. 12.]
[ 0. 7. 17.]
…
[ 1. 18. 62.]
[ 3. 21. 67.]
[ 5. 23. 69.]]]]
[7]: image = preprocess_input(image)
[8]: model = VGG16()
[9]: print(image)
[[[[ -92.939 -111.779 -123.68 ]
[ -91.939 -110.779 -122.68 ]
[ -85.939 -108.779 -122.68 ]
…
[ -28.939003 -88.779 -117.68 ]
[ -28.939003 -88.779 -117.68 ]
[ -28.939003 -88.779 -117.68 ]]
[[ -92.939 -111.779 -123.68 ]
[ -91.939 -110.779 -122.68 ]
[ -85.939 -108.779 -122.68 ]
3
…
[ -28.939003 -88.779 -117.68 ]
[ -28.939003 -88.779 -117.68 ]
[ -28.939003 -88.779 -117.68 ]]
[[ -92.939 -111.779 -123.68 ]
[ -91.939 -110.779 -122.68 ]
[ -85.939 -108.779 -122.68 ]
…
[ -28.939003 -88.779 -117.68 ]
[ -28.939003 -88.779 -117.68 ]
[ -28.939003 -88.779 -117.68 ]]
…
[[ -94.939 -111.779 -123.68 ]
[ -89.939 -111.779 -122.68 ]
[ -86.939 -109.779 -123.68 ]
…
[ -36.939003 -95.779 -120.68 ]
[ -35.939003 -94.779 -118.68 ]
[ -35.939003 -94.779 -118.68 ]]
[[ -91.939 -110.779 -122.68 ]
[ -91.939 -110.779 -122.68 ]
[ -86.939 -109.779 -123.68 ]
…
[ -39.939003 -96.779 -120.68 ]
[ -39.939003 -98.779 -123.68 ]
[ -36.939003 -95.779 -120.68 ]]
[[ -91.939 -110.779 -122.68 ]
[ -91.939 -110.779 -122.68 ]
[ -86.939 -109.779 -123.68 ]
…
[ -41.939003 -98.779 -122.68 ]
[ -36.939003 -95.779 -120.68 ]
[ -34.939003 -93.779 -118.68 ]]]]
[10]: yhat =model.predict(image)
1/1 [==============================] - 9s 9s/step
[11]: label=decode_predictions(yhat)
[12]: label=label[0][0]
[13]: print('%s (%.2f%%)' % (label[1], label[2]*100))
4
church (86.41%)
[14]: image=load_img('/home/rmdstic/Downloads/cat.jpeg', target_size=(224, 224))
[15]: print(image)
<PIL.Image.Image image mode=RGB size=224x224 at 0x7F5DD84A4400>
[16]: image = img_to_array(image)
[17]: print(image)
[[[ 1. 1. 1.]
[ 1. 1. 1.]
[ 1. 1. 1.]
…
[ 1. 1. 1.]
[ 1. 1. 1.]
[ 1. 1. 1.]]
[[ 1. 1. 1.]
[ 1. 1. 1.]
[ 1. 1. 1.]
…
[ 1. 1. 1.]
[ 1. 1. 1.]
[ 1. 1. 1.]]
[[ 1. 1. 1.]
[ 1. 1. 1.]
[ 1. 1. 1.]
…
[ 1. 1. 1.]
[ 1. 1. 1.]
[ 1. 1. 1.]]
…
[[ 89. 70. 56.]
[103. 84. 70.]
[120. 101. 87.]
…
[ 1. 1. 1.]
[ 1. 1. 1.]
[ 1. 1. 1.]]
[[ 93. 74. 60.]
[100. 81. 67.]
[130. 111. 97.]
5
…
[ 1. 1. 1.]
[ 1. 1. 1.]
[ 1. 1. 1.]]
[[ 83. 64. 50.]
[105. 86. 72.]
[ 93. 74. 60.]
…
[ 1. 1. 1.]
[ 1. 1. 1.]
[ 1. 1. 1.]]]
[18]: image = image.reshape((1, image.shape[0], image.shape[1], image.shape[2]))
[19]: image = preprocess_input(image)
[20]: print(image)
[[[[-102.939 -115.779 -122.68 ]
[-102.939 -115.779 -122.68 ]
[-102.939 -115.779 -122.68 ]
…
[-102.939 -115.779 -122.68 ]
[-102.939 -115.779 -122.68 ]
[-102.939 -115.779 -122.68 ]]
[[-102.939 -115.779 -122.68 ]
[-102.939 -115.779 -122.68 ]
[-102.939 -115.779 -122.68 ]
…
[-102.939 -115.779 -122.68 ]
[-102.939 -115.779 -122.68 ]
[-102.939 -115.779 -122.68 ]]
[[-102.939 -115.779 -122.68 ]
[-102.939 -115.779 -122.68 ]
[-102.939 -115.779 -122.68 ]
…
[-102.939 -115.779 -122.68 ]
[-102.939 -115.779 -122.68 ]
[-102.939 -115.779 -122.68 ]]
…
[[ -47.939003 -46.779 -34.68 ]
[ -33.939003 -32.779 -20.68 ]
[ -16.939003 -15.778999 -3.6800003]
6
…
[-102.939 -115.779 -122.68 ]
[-102.939 -115.779 -122.68 ]
[-102.939 -115.779 -122.68 ]]
[[ -43.939003 -42.779 -30.68 ]
[ -36.939003 -35.779 -23.68 ]
[ -6.939003 -5.7789993 6.3199997]
…
[-102.939 -115.779 -122.68 ]
[-102.939 -115.779 -122.68 ]
[-102.939 -115.779 -122.68 ]]
[[ -53.939003 -52.779 -40.68 ]
[ -31.939003 -30.779 -18.68 ]
[ -43.939003 -42.779 -30.68 ]
…
[-102.939 -115.779 -122.68 ]
[-102.939 -115.779 -122.68 ]
[-102.939 -115.779 -122.68 ]]]]
[21]: model = VGG16()
[22]: yhat = model.predict(image)
1/1 [==============================] - 2s 2s/step
[23]: label = decode_predictions(yhat)
[24]: label = label[0][0]
print('%s (%.2f%%)' % (label[1], label[2]*100))
Egyptian_cat (56.31%)
[25]: print(image)
[[[[-102.939 -115.779 -122.68 ]
[-102.939 -115.779 -122.68 ]
[-102.939 -115.779 -122.68 ]
…
[-102.939 -115.779 -122.68 ]
[-102.939 -115.779 -122.68 ]
[-102.939 -115.779 -122.68 ]]
[[-102.939 -115.779 -122.68 ]
[-102.939 -115.779 -122.68 ]
[-102.939 -115.779 -122.68 ]
…
7
[-102.939 -115.779 -122.68 ]
[-102.939 -115.779 -122.68 ]
[-102.939 -115.779 -122.68 ]]
[[-102.939 -115.779 -122.68 ]
[-102.939 -115.779 -122.68 ]
[-102.939 -115.779 -122.68 ]
…
[-102.939 -115.779 -122.68 ]
[-102.939 -115.779 -122.68 ]
[-102.939 -115.779 -122.68 ]]
…
[[ -47.939003 -46.779 -34.68 ]
[ -33.939003 -32.779 -20.68 ]
[ -16.939003 -15.778999 -3.6800003]
…
[-102.939 -115.779 -122.68 ]
[-102.939 -115.779 -122.68 ]
[-102.939 -115.779 -122.68 ]]
[[ -43.939003 -42.779 -30.68 ]
[ -36.939003 -35.779 -23.68 ]
[ -6.939003 -5.7789993 6.3199997]
…
[-102.939 -115.779 -122.68 ]
[-102.939 -115.779 -122.68 ]
[-102.939 -115.779 -122.68 ]]
[[ -53.939003 -52.779 -40.68 ]
[ -31.939003 -30.779 -18.68 ]
[ -43.939003 -42.779 -30.68 ]
…
[-102.939 -115.779 -122.68 ]
[-102.939 -115.779 -122.68 ]
[-102.939 -115.779 -122.68 ]]]]
[26]: image = load_img('/home/rmdstic/Downloads/dog.jpeg', target_size=(224, 224))
[27]: image = img_to_array(image)
[28]: print(image)
[[[ 24. 17. 11.]
[ 24. 17. 11.]
[ 24. 17. 11.]
…
8
[168. 111. 58.]
[163. 106. 53.]
[161. 103. 53.]]
[[ 24. 17. 11.]
[ 24. 17. 11.]
[ 24. 17. 11.]
…
[168. 111. 58.]
[163. 106. 53.]
[161. 103. 53.]]
[[ 24. 17. 11.]
[ 24. 17. 11.]
[ 24. 17. 11.]
…
[164. 107. 54.]
[160. 102. 52.]
[157. 99. 49.]]
…
[[ 89. 97. 116.]
[ 89. 97. 116.]
[ 89. 97. 116.]
…
[ 47. 68. 89.]
[ 49. 70. 91.]
[ 49. 70. 91.]]
[[ 81. 89. 108.]
[ 81. 89. 108.]
[ 82. 90. 109.]
…
[ 41. 63. 86.]
[ 43. 65. 88.]
[ 43. 65. 88.]]
[[ 81. 89. 108.]
[ 81. 89. 108.]
[ 82. 90. 109.]
…
[ 41. 63. 86.]
[ 43. 65. 88.]
[ 43. 65. 88.]]]
[29]: image = image.reshape((1, image.shape[0], image.shape[1], image.shape[2]))
9
[30]: image = preprocess_input(image)
[31]: print(image)
[[[[-92.939 -99.779 -99.68 ]
[-92.939 -99.779 -99.68 ]
[-92.939 -99.779 -99.68 ]
…
[-45.939003 -5.7789993 44.32 ]
[-50.939003 -10.778999 39.32 ]
[-50.939003 -13.778999 37.32 ]]
[[-92.939 -99.779 -99.68 ]
[-92.939 -99.779 -99.68 ]
[-92.939 -99.779 -99.68 ]
…
[-45.939003 -5.7789993 44.32 ]
[-50.939003 -10.778999 39.32 ]
[-50.939003 -13.778999 37.32 ]]
[[-92.939 -99.779 -99.68 ]
[-92.939 -99.779 -99.68 ]
[-92.939 -99.779 -99.68 ]
…
[-49.939003 -9.778999 40.32 ]
[-51.939003 -14.778999 36.32 ]
[-54.939003 -17.779 33.32 ]]
…
[[ 12.060997 -19.779 -34.68 ]
[ 12.060997 -19.779 -34.68 ]
[ 12.060997 -19.779 -34.68 ]
…
[-14.939003 -48.779 -76.68 ]
[-12.939003 -46.779 -74.68 ]
[-12.939003 -46.779 -74.68 ]]
[[ 4.060997 -27.779 -42.68 ]
[ 4.060997 -27.779 -42.68 ]
[ 5.060997 -26.779 -41.68 ]
…
[-17.939003 -53.779 -82.68 ]
[-15.939003 -51.779 -80.68 ]
[-15.939003 -51.779 -80.68 ]]
[[ 4.060997 -27.779 -42.68 ]
[ 4.060997 -27.779 -42.68 ]
10
[ 5.060997 -26.779 -41.68 ]
…
[-17.939003 -53.779 -82.68 ]
[-15.939003 -51.779 -80.68 ]
[-15.939003 -51.779 -80.68 ]]]]
[32]: model = VGG16()
[33]: yhat = model.predict(image)
1/1 [==============================] - 4s 4s/step
[35]: label = decode_predictions(yhat)
[36]: label = label[0][0]
[37]: print('%s (%.2f%%)' % (label[1], label[2]*100))
Pekinese (70.83%)
[ ]:
11
